dashboard/src/components/ChartView/ChartView.tsx

```diff
@@ -50,18 +53,34 @@ export default function ChartView() {
 
   const location = ReactRouter.useLocation();
 
+  const [pluginObj] = useState({
+    name: plugin?.split("-")[0],
+    version: plugin?.split("-")[1],
+  } as Plugin);
+
   // Fetch the selected/latest version on the initial load
   useEffect(() => {
     dispatch(
-      actions.charts.fetchChartVersion(chartCluster, chartNamespace, packageId, queryVersion),
+      actions.charts.fetchChartVersion(
+        chartCluster,
+        chartNamespace,
+        packageId,
+        pluginObj,
+        queryVersion,
+      ),
     );
     return;
-  }, [dispatch, packageId, chartNamespace, chartCluster, queryVersion]);
+  }, [dispatch, packageId, chartNamespace, chartCluster, queryVersion, pluginObj]);
 
   // Fetch all versions
   useEffect(() => {
-    dispatch(actions.charts.fetchChartVersions(chartCluster, chartNamespace, packageId));
-  }, [dispatch, packageId, chartNamespace, chartCluster]);
+    dispatch(
+      actions.charts.fetchChartVersions(chartCluster, chartNamespace, packageId, {
+        name: plugin?.split("-")[0],
```

Comment:
Don't you already have `pluginObj.name` here? (and `.version`)

Reply:
That's was the issue I was facing at when I started this PR, if `pluginObj` is added to dependency array of this useEffect it ended up in an endless loop :D However, now it's fixed with `plugin: getPluginFromString(plugin),`